src/commands/force/user/password/generate.ts

```diff
@@ -45,13 +69,44 @@ export class UserPasswordGenerateCommand extends SfdxCommand {
         const connection: Connection = await Connection.create({ authInfo });
         const org = await Org.create({ connection });
         const user: User = await User.create({ org });
-        const password = User.generatePasswordUtf8();
+        let passwordCondition: PasswordConditions;
+        if (connection.metadata) {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const list: any = await connection.metadata.list({ type: 'ProfilePasswordPolicy' });
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-explicit-any
+          const profPasswordPolicies: ProfilePasswordPolicy = JSON.parse(
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+            JSON.stringify(await connection.metadata.read('ProfilePasswordPolicy', [list.fullName]))
+          );
+          passwordCondition = {
+            length: profPasswordPolicies.minimumPasswordLength,
+            complexity: profPasswordPolicies.passwordComplexity,
+          };
+
+          if (!passwordCondition.length || !passwordCondition.complexity) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+            const orgPasswordPolicy: OrgPasswordPolicy = JSON.parse(
+              JSON.stringify(await connection.metadata.read('SecuritySettings', ['passwordPolicies']))
```

Comment:
since you could need both org and profile password policies, you might as well get them both, in parallel (Promise.all), before starting any of the logic.  The logic should be a little easier then.

Reply:
I thought of using promise.all but if we have ProfilePasswordPolicy however it overwrites the OrgPasswordPolicy. 
So I thought it's better to first find ProfilePasswordPolicy and if it's not present then OrgPasswordPolicy. 
Let me know what do you think of this?

Reply:
2 calls in parallel should be
as fast as doing 1 and not needing to do the 2nd
always faster than making a second trip when you find out you need the Org-level